screens/NativeModule.tsx

```diff
@@ -1,20 +1,131 @@
-import React from 'react';
-import { StyleSheet, Text, Dimensions } from 'react-native';
+import React, { useState, useEffect, useCallback } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
 import { colors } from '../style/styleVariables';
 import ContainerFull from '../components/ContainerFull';
+import Header from '../components/Header';
+import { Camera } from 'expo-camera';
+import CameraComponent from '../components/CameraComponent';
+import MorseForm from '../components/MorseForm';
+import { translateToMorse } from '../morseMessenger/morseFunctions';
+
+const TORCH_ON = 'torch';
+const TORCH_OFF = '';
 
 export default function NativeModule() {
-  return (
-    <ContainerFull bgColor={colors.second}>
-      <Text style={styles.header}>Native Module</Text>
-    </ContainerFull>
-  );
-}
+  const [hasPermission, setHasPermission] = useState(null);
+  const [flashMode, setFlashMode] = useState('');
+
+  const buttonCallback = useCallback(input => {
+    asyncFlash(input);
+  }, []);
+
+  // Get the permision
+  useEffect(() => {
+    const effect = async () => {
+      const {
+        status
+      }: { status: string } = await Camera.requestPermissionsAsync();
+      setHasPermission(status === 'granted');
+    };
+    effect();
+  }, []);
+
+  function torchOffAfter(time: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_OFF);
+        resolve();
+      }, time);
+    });
+  }
+  function torchOnAfter(time: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_ON);
+        resolve();
+      }, time);
+    });
+  }
+  function short(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(400)
+        .then(() => torchOffAfter(100))
+        .then(resolve);
+    });
+  }
+
+  function long(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(400)
+        .then(() => torchOffAfter(700))
+        .then(resolve);
+    });
+  }
 
-const styles = StyleSheet.create({
-  header: {
-    fontSize: 25,
-    fontWeight: '700',
-    color: colors.light
+  function pause(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve();
+      }, 1000);
+    });
   }
-});
+
+  // function letterPause(): Promise<any> {
+  //   return new Promise((resolve, reject) => {
+  //     setTimeout(() => {
+  //       resolve();
+  //     }, 1000);
+  //   });
+  // }
+
+  // function wordPause(): Promise<any> {
+  //   return new Promise((resolve, reject) => {
+  //     setTimeout(() => {
+  //       resolve();
+  //     }, 2000);
+  //   });
+  // }
+
+  // const longOrShort = {
+  //   pause: pause,
+  //   s: short,
+  //   l: long
+  // };
+
+  const asyncFlash = async (sentence: string): Promise<void> => {
```

Comment:
would be nice to always use `function` or `const` inside FunctionalComponent for consistency, here you have `const`, above you have `functions`. Hooks are a special case as it's common to just pass arrow there